### ISSUE TITLE
Calculation of gradients w.r.t. inputs during backpropagation for MultiLayerNetwork

### ISSUE THREAD

**Issue by darrenfoong** [deeplearning4j/deeplearning4j#1201]
The current implementation of the backpropagation algorithm for MultiLayerNetwork (https://github.com/deeplearning4j/deeplearning4j/blob/master/deeplearning4j-core/src/main/java/org/deeplearning4j/nn/multilayer/MultiLayerNetwork.java#L1190) computes the gradient of the error with respect to the weights and biases, but not the inputs to the network.

Computing the gradient of the error w.r.t the inputs will be useful in some applications, e.g. training embeddings (seems weird, but has been done in the Stanford neural network parser).
